### ISSUE TITLE
a newly created parameter is registered with a wrong address

### ISSUE THREAD

**Issue by matcham** [ossia/libossia#583]
open this basic patcher containing a device, a model and a parameter:
create a new parameter in the patcher myModel : the newly created parameter is registered at the root level :

ossia.device: node created myDevice:/toto
ossia.device: parameter created myDevice:/toto

it should be at : myDevice:/myModel/toto

**_this bug occurs only when the patch is opened for the first time after Max startup.
If the patch is closed and re-opened : it works as expected_**

here is video showing the issue [https://www.dropbox.com/s/v1uz662uno82m96/new%20param%20wrong%20address.mov?dl=0](url)

<pre><code>
----------begin_max5_patcher----------
2306.3oc6Zs0iixaF95L+JPbc1Q1ly8tVUoJU88IUodU0nUQNAOYXqCf.mYm
8a09eu9.FrI.gjIjNa6NRCA7A38848fe7gu+vJ2sEuQpcc9SNO4rZ02eX0JY
QhBV077J2C321Qw0xl4lS9Zw1u3tVUEi7FSVboydBaSN9.otDuinqO+3grbJ
gI6KpovRLa2KY462TQ1wTe63GCV6jDJtBiheDr1AgdD374tWSwQl98.aJUUD
6akD0Kw0ssCYoRohKoeJw07yRpZzsFkak6yYTxqjp5rhbiW9JWbYoQwqL5h.
Q9Rg7EEutsnrbUQv1hpHulo6eXao3Jtxy3Z9wJob69Vnua2qoHkTkeLS9lTE
9iGzhjzJzgwhNKLV5p6fST.Rhg9.wOd9RD0GD1Aoby7dZwt+MQBT.cgEkj7r
7xJRMImgYMBea0ojmwGorMOWjypy9CoH.EepAp+4FQbvJEJgT9+yUYXZqFru
JKsHWHDVlBQw5O2SNv.g1Ht1oLxVjiKGnybe.NtLRk0bk7X8VbkvRskRLbS4
N2EET6pZ6Gk7Lqo5xr77dnHqnb7Jqx1+xD8caAuxCS8tk0Tu4Xtp1MbmB1lZ
7q1nMCSoMwp1u92v4YGvLBKSYBPf1JI4Xth9R8tpBJ0ReU075.0jxcx2Q9ZV
J6E4GxzYf27rRsSjaqUNMaOolYWFCuu1tjZ12TftQQG21DDugQNTR4ZgcC3g
GY0r5WJ9ZcSC0NZl.PWBOyfZyDeVkOUBvdIAqxxYNEbg.+nBWLalUtPnQECj
OLTF3BAJWcfW+LhCjUDXTiQ5OOWcw+3gGz2r9J0cdlmZ7dxfJ+NJAW4v85FU
mQSqynXk1lH+I.LCc1DEGezfSGQ3FBIS5NP34aRcpILw+Wqu.LVAEQxeR7VL
bI5NiKv2KhnFdKLXwPjv6Dh7GTmZJOcwDPxYBdZFTLrIgwbBdPSBIqm.VBt+
APWsqhJOhmBeh7WLWEzsDSjp2npLXVtBxqd.80atBa4ptq3vANiQqQiksOOk
7lAW1k1ig4Pc1d89JFC55cSBg1hy2OJB5A+YJN5NEFEemwDv0hH9A2mwfPQ+
bQWoAWVb1Jn60fyLmsNe5p4pnY01LNT76MqhLixZwPDrwgl60.z6JNlyHUih
MAyhGWCi+P3LvF+ovFAlrtg6x4fH+6DDUwkRhS+EF6J42omVn2hRuCF7ggKi
N5ArzrY77uH5LnaH.oDx24.ygfYCPisPAAuaBccqGGMKerUYQpCh5GFZpKNV
sSatZV+VGa0IkTyxxaWcxm5j+dM7PVZYAOKPijfR7EKvbyxgh5dv4yyxbdoR
dzLkbz4DbXbhTVUw8vftmVHIOblRdiFtHxPvbQu3EVHf2DSHOCF2ngfllP4S
KjjKRnMO7KXAwO9TblmTr3Bwrrh9myL5IshPEMtlvR4SKjnqctmc1ukQJlcl
rvkTJBuvLBiaFC7Mxmh7W5zon4lKKYlCgAiT4OLdZIE8YG63MYrCOamPZiaX
uBUwPKHrO2Tfm5XUTkp1oXv+ckrAh7Fla.BISIobrkOLr9.WF84lwQSSt4dw
QCN2PS8HYKmTLmnLuyyyPcTJjwXgI56WJ4dlfGb7QmZJTeHGbEajc5F0l1uA
yXUYa4Sftt+o13h1b48zhsXZucDdn8d9gNgSdUgQy4XwXuyncS++8chXjWh7
tomGF0Bebgp2WxXOtiPoaEmYDspL5A4vsp3qp0+ALkVabPC.mAHfdxwN7P9p
s4xy.LF3vlL7hFcJ.QypsVwHWqW5omvE9zhosG0AjO3TzEF510TKsk+bJ4Y0
GlGmxiNCWK9QNk8tKpQEGt3OA0+67YUbyEZFG8vac3a+dQJgNnS5D1FwxNzY
cN6A2Z.7x6+COcVPTn77XoVaKOU7MzC9qSm0uNcV+5zYsXmNK04xpDWw+nLR
EOm2+Pbu0N6H39H2Ti9qf9krjrAQRxWHiihgWvBsV8n6z1Xn.OQ9OZuAKt7U
rVl8qY5e9vEaCBGb20GbIp+eDNniL5t44Qja69qlGLwAMbixBKTc.saNbw2J
BogPSFoxVXu+AJcu+bSZPfo2ufwlOh0L9OcJ0RWS0jOMdXDJWyUvBgyPxR5l
nzU+g7m42ANM.vmItPyihzOz1b6UGYQkydfl85XX4xziQn3i1iIXOVfmx.bb
1e8Y9ISUnFDrmaZG2oioYE+SIClM+NI+nJhSyyxFZ1tmyykxmgPwIDf0QOtp
ZaIapa6S7A0QI9PXrXlBdPuHTn7N9MAAlqpfpOPcm7C7S.HQS8CQQ9Ax6hQd
AdxobzqantuE.ln9BfjXfu5NdQP4jT55FNeeCisHCZGkUEkEUsTWezKos8GY
E6qvoYMa6HvJQ35FOqJds84d3dfCnY59X5I1ZL963CEGvadgytiJX3IRuWwk
PxTVE7tc7ulARCUSGKnaNY5H02o3ge9suwmccwW08nY0DrDGSl4Oo4lO3DVe
xZKfkiDH0hMY4hPDhk1DCiC8hLUo0y2or0fMrOFB3CU9XPT.LQ5hB.HOwcvw
8wBikPr4Om156uC4iQIC3SZrS4WmsRrlFmZqzpWPipXd2P9dL7VaQfPIGrce
Ew7vjnj0Cdmoo2pSgQIA.ooKHHFpxN.BghzLvy6kwcv3cBHsjmd2IpgLVoll
k1tT.e+DQ57Nt1NHmyI7RBggiEB+BgVJVrB9D91Rn5yK1vIVtDuCn2k5cDX5
7eSRO8kFZq.fVst5W0A+WwWOzzyxFEKFuR5KHu3KF9yZzq9fIZNNr8W3saDJ
JTcw2jOWBFOayEOvi0kVsXnvbTHevcoGOxOzG1bWDzCMZX9Deg9gSy7k2C4M
SxUyE5chTtVu3jPPTrL8crfOhzWFFDxCxuoVA0PqctxWMgrIG6ab77DBUqMt
.GaPtdMZFbiz50bAF0rG+KzijoSdchyof0YTnuLvKrkTYhnnaZVHCAD8QW.8
9.Kf+sJBI+hswJludh+jAl73RTzRIbnOxBm2GSg6eIm.wEaWi4bhjNdQb9TJ
1kAIA9ASlLGIZD.DuDJ.5mcEv6mPEnN6PIk7dI2wvkW.2teCyJb9MwrwcuFs
PspOiroaO7iG9O3IcyjK
-----------end_max5_patcher-----------
</code></pre>

**Comment by matcham** [comment]
I've just updated to latest release 1.0.8 after reporting this issue...
something has changed : 
now it works as expected only once after Max startup.
If the patch is closed and re-opened : the issue is back

**Comment by vincentgoudard** [comment]
Hey all,

adding my voice to this one. Just tested the v1.0.10 and parameter registration is still wrong.

There seem to be several issues related to this problem of device/model/parameter registration. 
To me, this is ultra-critical and a real stopper in using ossia (but I fear I'm not the only one...), so maybe we could unite efforts to get this solved ?

Having kinda made my own ossia style parameter registration in Vanilla Max,  I know one problem is about the cascade-initialization of device/model/parameter. I came up with something that could maybe be transposed to the OSSIA code (with which I am not ultra familiar tho).

@jcelerier @bltzr @avilleret : what do you think ?

**Comment by avilleret** [comment]
hi, 

> so maybe we could unite efforts to get this solved ?

sure ! help is welcome, what we need is to isolate edges cases like the one in this issue and try to find a way to fix it
but when I read :

> this bug occurs only when the patch is opened for the first time after Max startup.
> If the patch is closed and re-opened : it works as expected

I'm quite disappointed... and desperate... 

I will try to have a look at this one soon.
And I'm very interested in what you made @vincentgoudard 
To my eyes, one thing which is very hard is that I never find a good place describing how max object are instantiated.
And it seems that at least opening a patcher from Max and opening it by double clicking on it from Finder doesn't lead to the same behavior which is a big pain.

A lot of efforts have been made in ossia-max to make registration as fast as possible but it seems that those have very bad side effects...

**Comment by vincentgoudard** [comment]
The general rule in Max is that deeper level patchers are loadbanged first, and in a same patcher, it's based on the layer order (front-most loadbang is triggered last). I don't know how this connect with "instanciation" though,  but ultimately, I don't think it should rely on it either, because we can very well add/remove/rename any ossia device/model/parameter at any time (and I do, AFAIC).

What I am doing is I am using a global initialization flag registry, pulling the init signal from parent at instanciation and getting ready for a pushed init signal otherwise. So the initialization algorithm goes this way :
- check if parent context (a model or device in ossia) exists, and output 1 if it is flagged as initialized (a "pull" model, for instanciation)
- if not, set a receiver's name to receive from that parent, so that it will receives a init signal when parent is ready (a "push" model, for reactiveness)
 
But the difference with OSSIA is that here, I am explicitely giving the parent context to my parameters (either as argument, or setting it with a later message), while in OSSIA, I guess you need to retrive it by scanning the patcher structure, and is even more obfuscated by the fact that any OSSIA object can create the OSSIA global device (IIUC), so you don't know when it's ready.

A solution could possibly be have a deferlowed init signal sent by the OSSIA global device to all OSSIA component, so that we are sure that they are instanciated.

Please note that I am not at all familiar with OSSIA code, so I may very well be talking non-sense regarding what's really happening there... :)

**Comment by vincentgoudard** [comment]
BTW, I just tested Mathieu's example and it works fine here.
Will try to create a reproducible MWE showing the still-wrong bits.

**Comment by avilleret** [comment]
You have pretty good ideas @vincentgoudard but they have been almost already there :-)

Currently the registration mechanism of ossia-max object is quite cumbersome for several reasons.

When you add an object to a patch that already holds some ossia-max objects, then this object will look in parent patches until it find a node to connect to (ossia.model, ossia.device...). This works pretty well.
If the added object create node under which some other could register, then it will search for potential children and register them.

When you open a patch with ossia-max object, there is a lock that prevent every objects to register themselves. Because as you said, the deeper subpatcher is loadbanged first, and as explained above an ossia object need a parent object to register under.
So there is a struct that holds top most patches (the ones that don't have parent) and their loadbanged state.
When an object is loadbanged, if its top most patch is not loadbanged yet, then its registration is postpone to the next clock tick (after all objects have been loadbanged).

There are still improvement to be made and this is only an introduction to this big mess. 
Then there is pattern matching. And poly~ and bpatcher...
Welcome to Hell.

Also some issue are not reproductible between computer for example I can't reproduce https://github.com/ossia/libossia/issues/559 as you can't reproduce #583.

**Comment by vincentgoudard** [comment]
I am also using poly~, bpatchers and a regexp-made pattern matching in my own OSSIA ersatz, so yea, I know what you mean :'(
Well, let's tackle it!

What's the best place to discuss all those exquisite details ? Here ? Gitlab ? A new specific thread in issues ?

**Comment by vincentgoudard** [comment]
A simple (and reproducible, here) non-working example : 

Steps to repro : 
- Start Max (version 8.1.6 running here)
- Paste the code below in a new patcher
- send message `namespace` to the device
	- expected : `my_parameter` and `myModel/my_parameter`
	- obtained : `my_parameter`, `my_parameter.1` and `myModel/my_parameter`
- open the subpatcher "p model"
- send message `namespace` to the model => `my_parameter` = fine
- now delete the parameter in that subpatcher, then undo this action
- again, send message `namespace` to the model => parameter is gone!
- jump back to root patcher and ask the namespace to the device :
	- expected : same as expected previously
	- obtained : `my_parameter`, `my_parameter.1`, `my_parameter.2`


```
<pre><code>
----------begin_max5_patcher----------
2498.3oc6b0ziihqEccpeEnrNcIr4i.yp4IMqFo2ufVihHItRQODHBb5t5YT
+e+YrAhArM1Twjp0qWzcJrgv8d70Gewbt4eeZ058EugpV67aNe1Y0p+8oUqn
MU2vpliWs9bxaGxRpnm15bz2J1+k0aXcgQugoMew4DBuKO4Lp5RxATa+4WOm
lmgvzqE1zX5Q5kP9Z9TL2IVbE2dlflVYMg+9EDyDWu14uZ55RB9vqo4m1UhN
fY85u8Y2MNfH5GvnP5Gvmc6eMnxFuqw8Vs9kzLzWQkUoE4b27UqStbgq4UbW
RMl7kB5WTzltlRyYMA5ZpD80z1qOrq0jRhkiIl80Rpes9sP+029ZJNhJyulR
+lXM9imZMI53vMTt9hqGtZ69FV.CfT222s9CO+n5O7cCugFjA5SYEG9aDczv
sswhKn7z7KknJTNNA2X7cceD8Rx0L7tWJxwUo+C0D.02JA8+RiIJryZmfZ++
mxzjrNO3TY5wh7Zin2PQcys2tO6.BnCzA7NC8LxStH3hIAZDbQRmUDm7Z09j
x5Qp8YHt.UR3cQQV+t5ttLzK3ltujlmO.EwEWj2YY5oWUbs6KHcdV02Msmpc
WyY8tiDTf2Uk709nMNIKqY1Z+u92RxSOmfQ3T1P.zsqSTdBwQes5PYQVVO+k
0yWEzyQRP9Az2ROhekdi3CFHmd5k1fn0cixGSOgpv8aCmbppeKU3uy.ctltt
uYR7NL57kLhWz+DHSORqvUuV7splSrMPiG.tQ4wOolm5qW6pn.GPCVllicJH
FQxyLbg+z5wFB35fiQzavEvwJ5x0i.5uP5TcfKaxgq2M5u1K6GO8T6erYl9N
g4oJ4DRnyeHCkT5Ph5j5yPw9brbelGkjuZfLHAFw.iX5GAtVARTFNfH7MGcp
P35+YZrvVagKfHFTrk9Qr2iBW.lhHg1FQXqWFFrzHx+j4TkQnKT.IRl7DHGR
fJgjM5.Kr4OgMbJOxIPlFp.sVnBiGwigMa8sBlPcOotrqXWFXOWtMgKhe619
+8tpCEmOSxXr2pwTyK+H5MtbYscDC1IyYuwK5BdGSh1mjeRWDzkCB+YYdTjc
ipr67HMvDWiQDqstrevCcQn2Y5JvPKiKOpzUvN6c9jw4p.m6JyTBkM0qAf0N
C2l0jhVZn4Pw0bLpTJ1DHAa7kiM9pvlZLYSSxKZBQMo107P.gfkFhJINAxY3
Vioc9cfEHAu1Gcz6iStLd919gBceWoy.ui.DyGLNokfYuUAMq7F5duSn619w
kklKaWVnFTc+hglphqkGZGMaRq0ouscDUgSy61cxOealhitgtFaD0e4.MrBu
QV64ziWJHLUMvAvM54f1kwBia+aaZ25fdd.KhdPcMhfIQu.Jh0ri1P+tCrkk
6qooWmU3DldD0zgPpoG32cPuKpn7H6UT.dr9S3nyqyxbsiko6LLpK3oBo8.d
0fKHpYAM.6ECYuYYPcmkEOUPBLlFW.1xhU3NxVldntA3QSY6rPZulzHtcfsr
7s5Z4g1jZKxPtMqXET1asm7nbTzKfFxwdVhFNK5Q1zz0A.sJ9oMMuMsBsY.m
NGC5fHD3wd5v3tirnkaBIhULBcYx1ZQaPaNIMyVngMsMsGKRmFe2xQqYUKMy
QqowVQNrt9EYebG6k1uKAiKS2Sd.5pgp1vnWt7orh8IYCdivhd2yOcy3n+OC
X0QXL8eyn2d7ei0Di28WSLaYaw5fGu2PG7Ko3mOfxx1WqZDkNSYw2Xa9i6Xu
CztIgjGvLq2UIVRIh2bn0bRTvUFvjkV0ayhtATREfhbHDBoaPsG6CPnGmthH
dRm7Gf9tb2jARfo9LOhdg88RlhSd9uvM0ePeD7a+GKeUwM+IP6+b9qYMNJQA
WLQKbIojfKXToy4uu61A+dMr5TqsgeuQDOjgHOgAAfwi49hingRin2nUTsua
LamK19thqUhGLQbP.i+fWMGS5wAydNLIBgohrOSFyAv.+35A+swgtaip+qnX
e.HhFKngV37op.C3EXCD5hyM43HDXDL8GJFXbk6IdALWoIUF92A8++IquHF6
CjF3GxdOQ9Qt+RUe+RUe+RUe1SUeBE27iWdagLM95BkH3YQri8g.orjxXJGb
4hXLkvZJm4bZ1SYLn8XQ0fI0bQRKkRUCZUcnVmjdUSJVEzr5Q0NAc6jTtSR6
NA06zzuSRAqAMrNTwlPGqfRdRZY0TypomUSQqjlVFUsX5ZIT1ZQaqI0sX56g
DMCowG0+Tz4FHTakuBVUB1V8ahc9B2t2Nlze+ddu3zXAcOsntUtxmpU+TuBn
FqBNegdaQHbxPMYppxn3rsKANNsvve73HXtHX3Rhfx0v2iCAEJxbilLGnFBg
SBgazEFmT34eHlPO2PQ3hDJNoP0sHFNTzWJE9kJgraQ3YJQsqRGTRzB0iKpb
nv2MKIFvcXh8HgvOawv+y8b6nkK58wL2VSLzc1H3hjqyjhs+m9TFggKHN9QL
kwQB22L368lsiLg7Oew7+3fxwB8WoX+UJ3e0h9eVB+e9h++wAoxKL.ixIGrv
IkqnXA9nlOom+RtIDt2kTJgK.nNpvCLKowf28VeoQgHbeRBu+9bOp3DFCjhT
Ok.3bZUTYjRploZpLJDYtd1VC7LnNNlNJbaQ7rPC7rAZGbQru.SP9nGjABtq
gFZne0EwyFqKXCzF7xXg.CrvGlApczguNgGZHQ8Ew0Fq8eCz++xXgFwZG9Hr
vvYvtoN7X55PYY7LS3siMHU.kEGzx4ZFMm1ax4zZTyVKiuYBku3f2wUJ2GOK
WByvbpYQI0s3x3y287s0nBSWDOCXB8wvp2c4rPcYA7zKuuIpM5kwuL.3ASup
NWm7hUapBcYnR8lkfdTW3KhJ9k9Nfk+ICb1ZpzytZpjsHzV676ggnBoQsiOr
fZjVTMhJrFEEWi5MhUPQ1XXg1ntXalPVqdwxJ3F4EcirBuwdEeycJlPg5ime
g4nb27DUfNp283Y9SLCa3zuo9oFVrNKE90UHO+2hinLSAnf6BcyLpsG0fJno
1X7T+SElU9UYwvewCrSUaaVkiub+JanaszBBmpVZ2Bnoby1abtCdvHZ7nJUW
ncC7XoO6E0cj9+jhbKrkJ.YIwscKs7mImKNmr60zSDG9zq35bqJSx4y7nUH8
hAijCGP4XtIo.F4evsU.3Q8MbfbI4BEoR50mI2sTdG7Ce8NSQ89DErZDbXnT
SkBplXXZRgQ0hnt2GOMtO8sk9S5s5Mt27iQ6OUHc1.f8blPutiFaotuKKMVC
KsOEjXSExdOcbGzariExLnZgpsoAUGzfpBZb0.IuJfFV8OzYOC3EXXit7AC3
BZfPs4.Zl9JYte+48sHjjhS5oe7z+C.MCTEI
-----------end_max5_patcher-----------
</code></pre>
```

**Comment by jcelerier** [comment]
if there's a lot of various things to discuss, may make sense to make a specific project to list them all : https://github.com/ossia/libossia/projects

**Comment by matcham** [comment]
Hey
I'm happy (and reassured) to see that this topic brings attention :-)

about the example patches showing the issue :
after instaling latest release (1.0.10) , the first one I posted here misbehaves now only on mac here. On my windows PCs, it seems to be fixed...

about Vincent's patch : 
in your [p subpatcher] there is an 
ossia.device myModel
(and no ossia.model)
So you have 2 different devices, and I can't see why myDevice's namespace should contain myModel/my_parameter ...

here, myDevice's namespace is always
my_parameter, my_parameter.1
which demonstrates another issue : myDevice sees a parameter who belongs to another device, and I presume that it shouldn't (.. but maybe I'm wrong!)

I would be very happy to help fixing this mess because yes, it's ultra-critical - actually pacthing ossia-max requires closing and reopening the patch every time something changes in ossia namespace, and remotes randomly work #586
I hope I can be useful  with my Max-exclusive coding skills...

**Comment by vincentgoudard** [comment]
Yes! This is ultra-criiiiiiitiiiiicalllll! :) 
About my patch, sorry this is a simple copy/paste mistake, the device inside the subpatcher should actually be a model.
All the reported misbehaviours are actually happening.
I updated my previous message with the correct patch containing the model.

**Comment by avilleret** [comment]
@vincentgoudard with your patch I got the expected behavior when creating the patcher with File > New from Clipboard
but when pasting it in a new patcher, I got the same as you : an extra `my_parameter.1` is created.

be positive : I can reproduce the issue ! let's fix it :-)

**Comment by avilleret** [comment]
this should be fixed with f422cc5a in branch feature/581-namespace_object

but there are still edge case to explore, for example pasting an abstraction with model and parameter in subpatcher
if someone has some time to make some test around that

**Comment by avilleret** [comment]
for those who want to try but can't build here is a macOS package : https://framadrop.org/lufi/r/ZoMJJS0e3V#vfujQp4qPBq6mOdGfugMJJag5qenkJRza+DZBNcw0iU= (expire on Nov 7th 2020)

**Comment by matcham** [comment]
Thanks Antoine 
would love to give it a try, but I can't build... and with the package you sent, I get :

ø.remote: unable to load object bundle executable
 2020-10-09 09:52:23.086 Max[15742:501937] Error loading /Users/mathieuchamagne/Documents/Max 8/Packages/ossia/externals/ossia.remote.mxo/Contents/MacOS/ossia.remote:  dlopen(/Users/mathieuchamagne/Documents/Max 8/Packages/ossia/externals/ossia.remote.mxo/
 Contents/MacOS/ossia.remote, 262): Library not loaded: @rpath/librapidfuzz.0.0.0.dylib
   Referenced from: /Users/mathieuchamagne/Documents/Max 8/Packages/ossia/support/libossia-max.dylib
   Reason: image not found

Will have to wait for the next release.. :-s
+ my 2 big OSSIA-max projects are windows only ; any clue on the reason of Mac/windows discrepancy ? (with latest release, I couldn't reproduce the issue on windows...)

**Comment by avilleret** [comment]
oups we have to tackle this error... will try to send you an up-to-date package asap

**Comment by avilleret** [comment]
please use that link to download statically linked ossia-max : https://framadrop.org/lufi/r/vza4rByQkq#pG+5skTf6Q0lcrbJVEOc+ppMMFEDuwPN+msTb0XYTI8=

**Comment by avilleret** [comment]
concerning the discrepancy between macos and windows, I have no clue at the moment

**Comment by matcham** [comment]
Thanks Antoine

the latest package you sent do works on my Mac.

but : still have the same issue (the first one reported in this thread) : 
*only* the first time I open the patch, if I create a new parameter in a sub-model, the parameter is registered at root 
ossia.device: node created myDevice:/myParam1
instead of 
ossia.device: parameter created myDevice:/myModel/myParam1.1

> And it seems that at least opening a patcher from Max and opening it by double clicking on it from Finder doesn't lead to the same behavior which is a big pain.

I can see 5 scenarii possible to open a patch : 
- start Max, then File/Open... (or Open Recent Patcher) (I presume that load -> [pcontrol] does the same...)
- start max, then double click a patcher in the finder
- double click a patcher in the finder (--> start Max automatically)
- start Max, then File / New from clipboard (I guess it's the same than create an empty patcher then paste content of the clipboard)
- instantiate the patcher in a poly~

I also suspected that the method used to open the patch would lead to different results... 
but after many tests.. it seems to me that it's totally random :-s

I've made this automatic tester patch that load the test patch automatically and makes stats.
https://www.dropbox.com/s/orozccexvthdonw/new_parameter_wrong_address_tester.zip?dl=0
(I also suspected that the wait time between opening the patch and creating a new parameter has some influence on the result.. but finally it appears that .. no so much!)

In my tests here, it works fine ~ 65% of the time (Mac, Max 8.1.7)
(which is not so bad after all :-) but I'm sure it can be improved !

**Comment by vincentgoudard** [comment]
On top of that, one thing that will probably affect the behaviour is to
create ossia objects after some others have already been created (and
possibly deleted), as the ossia global device and Max memory may not be
clean anymore.

**Comment by matcham** [comment]
please note that when the patch is loaded and the issue occurs, it will always fail (you can delete the parameter and create it again, it will always have a wrong address.)
and when it works as expected, it seems to work fine everywhere in the patch as long as it stays opened.
